bigquery/src/main/java/org/apache/iceberg/gcp/bigquery/BigQueryTableOperations.java

```diff
@@ -0,0 +1,296 @@
+    addConnectionIfProvided(tableBuilder, metadata.properties());
+
+    client.create(tableBuilder);
+  }
+
+  private void addConnectionIfProvided(Table tableBuilder, Map<String, String> metadataProperties) {
+    if (metadataProperties.containsKey(TABLE_PROPERTIES_BQ_CONNECTION)) {
+      tableBuilder
+          .getExternalCatalogTableOptions()
+          .setConnectionId(metadataProperties.get(TABLE_PROPERTIES_BQ_CONNECTION));
+    }
+  }
+
+  /** Update table properties with concurrent update detection using etag. */
+  private void updateTable(
+      String oldMetadataLocation, String newMetadataLocation, TableMetadata metadata) {
+    Table table = client.load(this.tableReference);
+    if (table.getEtag().isEmpty()) {
+      throw new ValidationException(
+          "Etag of legacy table %s is empty, manually update the table via the BigQuery API or"
+              + " recreate and retry",
+          tableName());
+    }
+    ExternalCatalogTableOptions options = table.getExternalCatalogTableOptions();
+    addConnectionIfProvided(table, metadata.properties());
+
+    // If `metadataLocationFromMetastore` is different from metadata location of base, it means
+    // someone has updated metadata location in metastore, which is a conflict update.
+    String metadataLocationFromMetastore =
+        options.getParameters().getOrDefault(METADATA_LOCATION_PROP, "");
+    if (!metadataLocationFromMetastore.isEmpty()
+        && !metadataLocationFromMetastore.equals(oldMetadataLocation)) {
+      throw new CommitFailedException(
+          "Cannot commit base metadata location '%s' is not same as the current table metadata location '%s' for"
+              + " %s.%s",
+          oldMetadataLocation,
+          metadataLocationFromMetastore,
+          tableReference.getDatasetId(),
+          tableReference.getTableId());
+    }
+
+    options.setParameters(buildTableParameters(newMetadataLocation, metadata));
+    try {
+      client.update(tableReference, table);
+    } catch (ValidationException e) {
+      if (e.getMessage().toLowerCase(Locale.ENGLISH).contains("etag mismatch")) {
+        throw new CommitFailedException(
+            "Updating table failed due to conflict updates (etag mismatch). Retry the update");
+      }
+
+      throw e;
+    }
+  }
+
+  // To make the table queryable from Hive, the user would likely be setting the HIVE_ENGINE_ENABLED
+  // parameter.
+  //
+  // TODO: We need to make a decision on how to make the table queryable from Hive.
+  // (could be a server side change or a client side change - that's TBD).
+  private Table makeNewTable(TableMetadata metadata, String metadataFileLocation) {
+    return new Table()
+        .setExternalCatalogTableOptions(
+            BigQueryMetastoreUtils.createExternalCatalogTableOptions(
+                metadata.location(),
+                buildTableParameters(metadataFileLocation, metadata),
+                hiveEngineEnabled(metadata)));
```

Comment:
Do we actually need to reference the `HIVE_ENGINE_ENABLED`?  It looks like this is an unnecessary hold-over from hive.  If we get rid of this, I think you can also get rid of references to the Hadoop Configuration (I think this is the only thing we're referencing from it).

Reply:
We plan to support https://iceberg.apache.org/docs/latest/hive/#create-external-table-overlaying-an-existing-iceberg-table for our users. BQMS is not just BigQuery also become a catalog implementation for other engines.

Reply:
I don't know what that means, and my cursory look into it, I'm not sure it's safe.  I'd rather we start without this complexity so we can get a clean implementation in and then worry about adding this in the future.

Reply:
Thank you for comment. I removed hive support from the catalog.